onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -224,6 +224,14 @@ def retrieve(self, request, *args, **kwargs):
 
     def create(self, request, *args, **kwargs):
         """Create and cache user profile"""
+        invitation_id = request.query_params.get("invitation_id")
+        invitation_token = request.query_params.get("invitation_token")
+        request.data.update(
```

Comment:
can we add a condition here if invitation_id and invitation_token are passed as query params before doing the update?